tests/integration_tests/functional/test_cpu_features.py

```diff
@@ -142,7 +144,100 @@ def test_brand_string(test_microvm_with_api, network_config):
 @pytest.mark.skipif(
     PLATFORM != "x86_64", reason="CPU features are masked only on x86_64."
 )
-@pytest.mark.parametrize("cpu_template", ["T2", "C3"])
+@pytest.mark.skipif(
+    cpuid_utils.get_cpu_vendor() != cpuid_utils.CpuVendor.INTEL,
+    reason="CPU templates are only available on Intel.",
+)
+@pytest.mark.skipif(
+    utils.get_kernel_version(level=1) not in ["5.10", "4.14"],
+    reason="Supported kernels are 5.10 and 4.14",
+)
+@pytest.mark.parametrize("cpu_template", ["T2S"])
```

Comment:
Testing the MSRs not useful for other templates?

Reply:
It should be. Though since `T2` and `C3` did not set the `ARCH_CAPABILITIES` MSR, the value of that MSR is different between different CPU types (see https://github.com/firecracker-microvm/firecracker/pull/3066).
So we can either:
- add an exceptions for different template types when checking values
- deprecate `T2` and `C3` since they are somewhat broken IMO

I wanted to keep the more generic implementation in this PR and follow up with another once we made the decision.
I can add a `TODO` here so that we don't forget about it.

Reply:
I like the idea of deprecating `T2` and `C3`. Hopefully we can deprecate all hard-coded templates soon as well..

Reply:
Yes, I added a TODO in the comment to follow up on this when we have a decision.